File: Hackerrank/30_days_of_code_cpp/day_28.cpp
#include <regex>
#include <vector>
#include <iostream>

using namespace std;

int main(){
    int n;
    vector<string> names;
    regex domain ("@gmail\\.com$");
    cin >> n;
    for (int i = 0; i < n; i++) {
        string name;
        string email;
        cin >> name >> email;
        if (regex_search(email, domain)) {
            names.push_back(name);
        }
    }
    sort(names.begin(), names.end());
    for (int i = 0; i < names.size(); i++) {
        cout << names[i] << endl;
    }
    return 0;
}

File: Hackerrank/30_days_of_code_csharp/day_28.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text.RegularExpressions;

class Solution {

    static void Main(String[] args) {
        ArrayList names = new ArrayList();
        Regex tester = new Regex(@".+@gmail\.com", RegexOptions.IgnoreCase);
        int n = Convert.ToInt32(Console.ReadLine());

        for(int i = 0; i < n; i++){
            string[] entry = Console.ReadLine().Split(' ');
            if (tester.Match(entry[1]).Success) names.Add(entry[0]);
        }

        names.Sort();

        foreach (string name in names) {
            Console.WriteLine(name);
        }
    }
}
